core/tests/test_utils.py

```diff
@@ -809,9 +942,22 @@ def __get__(self, instance, owner):
         return self
 
     def pre_call_hook(self, args):
+        """Override this to do tasks that should be executed before the
+        actual function call.
+
+        Args:
+            args: list(str). Set of arguments that the function accepts.
+        """
         pass
 
     def post_call_hook(self, args, result):
+        """Override this to do tasks that should be executed after the
+        actual function call.
+
+        Args:
+            args: list(str). Set of arguments that the function accepts.
+            result: str. Result returned from the function.
```

Comment:
Ditto, why are these both necessarily str? Please explain the reasoning.

Reply:
Sorry, but there types are-- list(*) and * respectively as this is the function called after many other functions of many types.